lib/communicator.js

```diff
@@ -16,8 +16,12 @@ module.exports = class Communicator {
     this.url = removeTrailingSlash(url)
   }
 
-  async post (path, body) {
-    return this.call('POST', path, JSON.stringify(body), { 'Content-Type': 'application/json' })
+  async post (path, body, requestOptions) {
+    const headers = { 'Content-Type': 'application/json' };
+    if (!!requestOptions && requestOptions.hasIdempotencyKey()) {
+      headers['Idempotency-Key'] = requestOptions.getIdempotencyKey()
```

Comment:
it should be simple JS object, passed without constructor. 

JS developers are not used to use verbose class, constructors and DTOs for passing simple parameters.

Reply:
done.